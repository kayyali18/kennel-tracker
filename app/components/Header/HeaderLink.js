import { Link } from 'react-router-dom'
import styled from 'styled-components'

export default styled(Link)`
  display: inline-flex;
  padding: 0.25em 2em;
  margin: calc(10px + 6 * ((100vw - 320px) / 680));
  text-decoration: none;
  border-radius: 8px;
  -webkit-font-smoothing: antialiased;
  -webkit-touch-callout: none;
  user-select: none;
  cursor: pointer;
  outline: 0;
  font-family: 'Helvetica Neue', Helvetica, Arial, sans-serif;
  font-weight: bold;
  font-size: calc(16px + 6 * ((100vw - 320px) / 680));
  border: none;
  color: #ff4a6e;

  &:active {
    background: red;
    color: #fff;
  }

  &:hover {
    background: linear-gradient(#ff700b, #ff275f);
    color: #fff;
  }
`
