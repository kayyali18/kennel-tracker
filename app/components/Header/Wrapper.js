import styled from 'styled-components';

const Wrapper = styled.nav`
  display: flex;
  justify-content: space-around;
  flex-wrap: wrap;
  background-color: #fff;
  height: 100px;
  position: sticky;
  top: 0;
`;

export default Wrapper;
