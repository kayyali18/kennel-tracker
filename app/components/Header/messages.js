/*
 * Header Messages
 *
 * This contains all the text for the Header component.
 */

import { defineMessages } from 'react-intl';

export const scope = 'app.components.Header';

export default defineMessages({
  home: {
    id: `${scope}.home`,
    defaultMessage: 'Home',
  },
  newBooking: {
    id: `${scope}.newBooking`,
    defaultMessage: 'New Booking',
  },
  calendar: {
    id: `${scope}.calendar`,
    defaultMessage: 'Calendar',
  },
  search: {
    id: `${scope}.search`,
    defaultMessage: 'Search',
  },
});
