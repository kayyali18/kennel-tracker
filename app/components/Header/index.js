import React from 'react'
import { FormattedMessage } from 'react-intl'

import Wrapper from './Wrapper'
import HeaderLink from './HeaderLink'
import messages from './messages'

function Header() {
  return (
    <Wrapper>
      <HeaderLink to="/">
        <FormattedMessage {...messages.home} />
      </HeaderLink>
      <HeaderLink to="/booking">
        <FormattedMessage {...messages.newBooking} />
      </HeaderLink>
      <HeaderLink to="/calendar">
        <FormattedMessage {...messages.calendar} />
      </HeaderLink>
      <HeaderLink to="/profile">
        <FormattedMessage {...messages.search} />
      </HeaderLink>
    </Wrapper>
  )
}

export default Header
