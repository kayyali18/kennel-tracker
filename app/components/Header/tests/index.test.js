import React from 'react'
import { shallow } from 'enzyme'
import 'jest-styled-components'

import Header from '../index'

describe('<Header />', () => {
  it('should match snapshot', () => {
    const renderedComponent = shallow(<Header />)
    expect(renderedComponent).toMatchSnapshot()
  })
})
