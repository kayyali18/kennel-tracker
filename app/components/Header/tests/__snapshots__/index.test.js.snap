// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should match snapshot 1`] = `
<Wrapper>
  <HeaderLink
    to="/"
  >
    <FormattedMessage
      defaultMessage="Home"
      id="app.components.Header.home"
      values={Object {}}
    />
  </HeaderLink>
  <HeaderLink
    to="/booking"
  >
    <FormattedMessage
      defaultMessage="New Booking"
      id="app.components.Header.newBooking"
      values={Object {}}
    />
  </HeaderLink>
  <HeaderLink
    to="/calendar"
  >
    <FormattedMessage
      defaultMessage="Calendar"
      id="app.components.Header.calendar"
      values={Object {}}
    />
  </HeaderLink>
  <HeaderLink
    to="/profile"
  >
    <FormattedMessage
      defaultMessage="Search"
      id="app.components.Header.search"
      values={Object {}}
    />
  </HeaderLink>
</Wrapper>
`;
